packages/js/src/generators/setup-verdaccio/generator.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
+import { Tree, readJson, readProjectConfiguration } from '@nx/devkit';
+
+import generator from './generator';
+import { SetupVerdaccioGeneratorSchema } from './schema';
+import { PackageJson } from 'nx/src/utils/package-json';
+
+describe('setup-verdaccio generator', () => {
+  let tree: Tree;
+  const options: SetupVerdaccioGeneratorSchema = { skipFormat: false };
+
+  beforeEach(() => {
+    tree = createTreeWithEmptyWorkspace();
+  });
+
+  it('should create project.json if it does not exist', async () => {
+    await generator(tree, options);
+    const config = readJson(tree, 'project.json');
+    expect(config).toEqual({
+      name: 'test-name',
+      $schema: 'node_modules/nx/schemas/project-schema.json',
+      targets: {
+        'local-registry': {
+          executor: '@nx/js:verdaccio',
+          options: {
+            port: 4873,
+            config: '.verdaccio/config.yml',
+          },
+        },
+      },
+    });
+  });
+
+  it('should add local-registry target to project.json if it does exist', async () => {
```

Comment:
Can we add a test for not overwriting existing targets?

Reply:
added